src/scripts.js

```diff
@@ -1,24 +1,41 @@
 import './css/base.scss';
 import './css/styles.scss';
 
-import recipeData from './data/recipes';
-import ingredientData from './data/ingredients';
-import users from './data/users';
-
 import Pantry from './pantry';
 import Recipe from './recipe';
 import User from './user';
 import Cookbook from './cookbook';
 
+let users = fetch("https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/users/wcUsersData")
+  .then(response => response.json())
+  .then(data => data.wcUsersData)
+  .catch(error => error.message)
+
+let ingredientData = fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/ingredients/ingredientsData')
+  .then(response => response.json())
+  .then(data => data.ingredientsData)
+  .catch(error => console.log('ingredientsData error'));
+
+let recipeData = fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/recipes/recipeData')
+  .then(response => response.json())
+  .then(data => data.recipeData)
+  .catch(error => console.log('recipesData error'));
+//
+Promise.all([recipeData, ingredientData, users])
+  .then(data => {
+    recipeData = data[0];
+    ingredientData = data[1];
+    users = data[2];
+    onStartup(users, ingredientData, recipeData)
+  })
+  .catch(error => console.log(error.message))
+
```

Comment:
Not sure I 100% understand what this is doing. Would you mind explaining?

Reply:
So for each of the fetchs, in order for the information they retrieve to be used it has to wait for the promise to resolve.  Since those run asynchronously, there can be an issue with when the promise is resolved versus when that information is trying to be used by the javscript reader.  So to my understanding, promise all tells it to wait for all the promises specified to resolve before proceeding with the following functions.